Sources/AuthFoundation/User Management/Credential.swift

```diff
@@ -19,18 +19,30 @@ import FoundationNetworking
 /// Convenience object that provides methods and properties for using a user's authentication tokens.
 ///
 /// Once a user is authenticated within an application, the tokens' lifecycle must be managed to ensure it is properly refreshed as needed, is stored in a secure manner, and can be used to perform requests on behalf of the user. This class provides capabilities to accomplish all these tasks, while ensuring a convenient developer experience.
-public final class Credential: Equatable, OAuth2ClientDelegate {
+public final class Credential: Sendable, Equatable, OAuth2ClientDelegate {
     /// The current or "default" credential.
     ///
     /// This can be used as a convenience to store a user's token within storage, and to access the user in a safe way. If the user's token isn't stored, this will automatically store the token for later use.
     public static var `default`: Credential? {
-        get { coordinator.default }
-        set { coordinator.default = newValue }
+        get {
+            withAsyncGroup { @CredentialActor in
```

Comment:
Why in some cases we use Lock objects and in other cases are calling `withAsyncGroup`?

Reply:
This function is renamed in a subsequent PR to `withIsolationSync` and `withIsolationSyncThrowing`, which more accurately describes what's going on.

The goal isn't to arbitrarily lock a process or a thread, but rather to interact with a property or synchronous function within an actor in a different isolation context, which forces the call to be `async`.  When performing this access from a `nonisolated` property or function, we still need to ensure that the call to the underlying actor is made within the appropriate isolation context, while ensuring task cancellation and priority inheritance is respected.

Since we do this in a number of places, I abstracted this logic out into a reusable `inlinable` function.